src/collate.jl

```diff
@@ -0,0 +1,58 @@
+
+
+"""
```

Comment:
I don't recall if this has been discussed before, but is there a meaningful difference between `collate` and `batch`? If so, could the former delegate to the latter by default?

Reply:
yes it seems we are trying to replicate `batch` here

Reply:
yes it seems we are trying to replicate `batch` here